### PR TITLE
cmake-devel-gui: fix CMAKE_ROOT for cmake-gui

### PR DESCRIPTION
**Pull request by nilason**
#### Description

CMake code for `cmake-devel-gui` looks inside app bundle for CMAKE_ROOT, this patches it to look in MacPorts $prefix.

To start with, update `cmake-devel` only. Once it's been thoroughly validated in the wild, it can be integrated into `cmake`.

See: https://trac.macports.org/ticket/62855

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.6 21G646 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by mascguy** [comment]
My concern with this change, is that it could be a pain to update the patch with each update. So ideally this is something that should be submitted to upstream.

Nicklas, is that should you can do? (And be sure to reference this PR, so we know when it's been done.)

**Comment by nilason** [comment]
> My concern with this change, is that it could be a pain to update the patch with each update. So ideally this is something that should be submitted to upstream.
> 
> Nicklas, is that should you can do? (And be sure to reference this PR, so we know when it's been done.)

Not easy to find a simple yet universally workable/acceptable solution. The only one I could come up with is to set the path to "CMAKE_ROOT" (note that variable is a CMake variable only, not environment variable), where Modules etc are located, during compilation into the Info.plist. That env var can then be read in run time:

```diff

---
 Source/QtDialog/Info.plist.in |  5 +++++
 Source/cmSystemTools.cxx      | 10 +++++++++-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/Source/QtDialog/Info.plist.in b/Source/QtDialog/Info.plist.in
index 00a27c39c1..1334f9bb07 100644
--- a/Source/QtDialog/Info.plist.in
+++ b/Source/QtDialog/Info.plist.in
@@ -30,5 +30,10 @@
 	<string>${MACOSX_BUNDLE_COPYRIGHT}</string>
 	<key>NSHighResolutionCapable</key>
 	<true/>
+	<key>LSEnvironment</key>
+		<dict>
+			<key>MAC_CMAKE_INSTALL_PREFIX</key>
+			<string>${CMAKE_INSTALL_PREFIX}</string>
+		</dict>
 </dict>
 </plist>
diff --git a/Source/cmSystemTools.cxx b/Source/cmSystemTools.cxx
index 1fb0079103..3eb12c2b55 100644
--- a/Source/cmSystemTools.cxx
+++ b/Source/cmSystemTools.cxx
@@ -2592,11 +2592,19 @@ void cmSystemTools::FindCMakeResources(const char* argv0)
   }
   if (cmSystemTools::GetFilenameName(exe_dir) == "MacOS") {
     // The executable is inside an application bundle.
-    // Look for ..<CMAKE_BIN_DIR> (install tree) and then fall back to
+    // Look for ..<CMAKE_BIN_DIR> (install tree),
+    // alternatively look for the environment variable
+    // MAC_CMAKE_INSTALL_PREFIX set in Info.plist and then fall back to
     // ../../../bin (build tree).
+    const char* env_p = std::getenv("MAC_CMAKE_INSTALL_PREFIX");
     exe_dir = cmSystemTools::GetFilenamePath(exe_dir);
     if (cmSystemTools::FileExists(exe_dir + CMAKE_BIN_DIR "/cmake")) {
       exe_dir += CMAKE_BIN_DIR;
+    } else if (env_p &&
+               cmSystemTools::FileExists(std::string(env_p) +
+                                         CMAKE_BIN_DIR "/cmake")) {
+      exe_dir = cmSystemTools::GetFilenamePath(std::string(env_p) +
+                                               CMAKE_BIN_DIR "/cmake");
     } else {
       exe_dir = cmSystemTools::GetFilenamePath(exe_dir);
       exe_dir = cmSystemTools::GetFilenamePath(exe_dir);
-- 
```

**Comment by pmetzger** [comment]
Is this ready to merge? It's been waiting for a long time.

**Comment by mascguy** [comment]
> Is this ready to merge? It's been waiting for a long time.

I'm reluctant to merge anything that impacts `cmake`, given that it's a critical toolchain component. Particularly without a thorough review, and blessing, from @michaelld.

That being said, if this PR is changed to only modify `cmake-devel`, I'm more than willing to approve and merge. Whether it's ultimately added to `cmake` is up to Michael though, as he's the sole gatekeeper of that port.

**Comment by mascguy** [comment]
Nils, would you mind dropping the commit for `cmake` at this point? That'll give us a chance to test this in the wild via `cmake-devel`.

**Comment by nilason** [comment]
> Nicklas, would you mind dropping the commit for `cmake` at this point? That'll give us a chance to test this in the wild via `cmake-devel`.

No objections by me, Christopher. I can do it tomorrow, or if you have the time please go ahead (just change the commit message, to not close the ticket).

**Comment by mascguy** [comment]
> > Nicklas, would you mind dropping the commit for `cmake` at this point? That'll give us a chance to test this in the wild via `cmake-devel`.
> 
> No objections by me, Christopher. I can do it tomorrow, or if you have the time please go ahead (just change the commit message, to not close the ticket).

All set, I'll merge once CI completes. Thanks as always for your contribution Nicklas!

**Review by mascguy — APPROVED** [review]

**Pull request merged**
